src/include/recorder.h

```diff
@@ -0,0 +1,125 @@
+#include <fstream>
+#include <vector>
+#include <mutex>
+#include <chrono>
+#include "debug.h"
+
+// API opcode covered by rccl replayer
+typedef enum {
+  rrBroadcast,
+  rrReduce,
+  rrAllGather,
+  rrReduceScatter,
+  rrAllReduce,
+  rrSend,
+  rrRecv,
+  rrAllToAll,
+  rrAllToAllv,
+  rrGather,
+  rrScatter,
+  rrBcast,
+  rrGroupStart,
+  rrGroupEnd,
+  rrGroupSimulatedEnd,
+  rrCommInitDev,
+  rrCommInitRank,
+  rrCommInitAll,
+  rrCommInitRankConfig,
+  rrCommSplit,
+  rrCommFinalize,
+  rrCommDestroy,
+  rrCommAbort,
+  rrCommRegister,
+  rrCommDeregister,
+  rrMemAlloc,
+  rrMemFree,
+  rrRedOpCreatePreMulSum,
+  rrRedOpDestroy,
+  rrOtherCall
+} rcclCall_t;
+
+extern const char* rcclCallStr[];
```

Comment:
This should be defined within the header, so that Replayer can have access to it, without having to link in recorder.cc